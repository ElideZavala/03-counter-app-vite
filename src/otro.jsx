import PropsTypes from "prop-types";

const array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

const elementList = [
  {
    id: 1,
    name: "Elide",
    age: 30,
    country: "Mexico",
  },
  {
    id: 2,
    name: "Maria",
    age: 25,
    country: "Venezuela",
  },
];

// eslint-disable-next-line react/prop-types
export const Otro = ({ title, subtitule, name }) => {
  if (!title) {
    throw new Error("El titulo es necesario");
  }

  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h3>{name}</h3>
      <p>{subtitule}</p>
      <p>{subtitule}</p>
      <ul>
        {array.map((item, index) => (
          <li key={index}>{item}</li>
        ))}
      </ul>
      <ul>
        {elementList.map((item, index) => (
          <li key={index}>
            <p>{item.name}</p>
            <p>{item.age}</p>
            <p>{item.country}</p>
          </li>
        ))}
      </ul>
    </>
  );
};

Otro.propTypes = {
  title: PropsTypes.string.isRequired,
  subtitule: PropsTypes.string,
  name: PropsTypes.string,
};

Otro.defaultProps = {
  title: "",
  subtitule: "No tenemos nigun subtitulo",
  name: "Elide Zavala Vinagre",
};
