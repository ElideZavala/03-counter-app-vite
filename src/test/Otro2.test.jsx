import { render, screen } from "@testing-library/react";
import { Otro } from "../otro";

/* eslint-disable no-undef */
describe("Pruebas en <Otro/>", () => {
  const title = "Hola soy Goku";
  const subtitule = "Soy un subtitulo";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<Otro title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el mensaje "Hola soy Goku"', () => {
    render(<Otro title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe de mostrar el titulo en un h1", () => {
    render(<Otro title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<Otro title={title} subtitule={subtitule} />);
    expect(screen.getAllByText(subtitule).length).toBe(2);
  });
});
