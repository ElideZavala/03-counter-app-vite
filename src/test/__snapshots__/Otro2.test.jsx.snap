// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Otro/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Goku
  </h1>
  <h3>
    Elide Zavala Vinagre
  </h3>
  <p>
    No tenemos nigun subtitulo
  </p>
  <p>
    No tenemos nigun subtitulo
  </p>
  <ul>
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3
    </li>
    <li>
      4
    </li>
    <li>
      5
    </li>
    <li>
      6
    </li>
    <li>
      7
    </li>
    <li>
      8
    </li>
    <li>
      9
    </li>
    <li>
      10
    </li>
  </ul>
  <ul>
    <li>
      <p>
        Elide
      </p>
      <p>
        30
      </p>
      <p>
        Mexico
      </p>
    </li>
    <li>
      <p>
        Maria
      </p>
      <p>
        25
      </p>
      <p>
        Venezuela
      </p>
    </li>
  </ul>
</div>
`;
